### PR TITLE
Tidying the search function

### PR DESCRIPTION
**Pull request by gareth-j**
This massively simplifies the `search` function. It now:

- Uses the metadata store
- Checks dates only if a date is passed in
- Performs no ranking search

### Limitations

- Can still only search for a single data type, do we want to support searching across multiple data types?

### Closes

- #282 - *does this close this one, or does the original filename need to be added?*
- #362 
- #366
- #405 
- #422

### CODE REVIEW

**Comment by gareth-j** [comment]
mypy fails on the 3.10 runner due to a bug in mypy, see https://github.com/python/mypy/issues/13627.

**Comment by gareth-j** [comment]
This will close #411.

File: openghg/retrieve/_search.py [hunk 1]
```diff
@@ -102,7 +156,9 @@ def search_surface(
         start_date=start_date,
         end_date=end_date,
         data_source=data_source,
-        **kwargs,
+        network=network,
+        sampling_height=sampling_height,
+        # **kwargs,
     )
 
     # TODO - remove this cast once we've updated search to ensure return of SearchResults object
```
Comment (rt17603) [inline openghg/retrieve/_search.py:164]: Just noticed this comment - can the `cast()` be removed now?

File: openghg/dataobjects/_searchresults.py [hunk 2]
```diff
@@ -95,290 +88,107 @@ def from_json(cls: Type[T], data: Union[bytes, str]) -> T:
...
-            return None
+        return cls(keys=loaded["keys"], metadata=loaded["metadata"])
 
-    def metadata(self, site: str, species: str, inlet: Optional[str] = None) -> Optional[Dict]:
-        """Return the metadata for the specified site and species
+    def retrieve(
+        self,
+        dataframe: Optional[DataFrame] = None,
+        sort: bool = True,
+        elevate_inlet: bool = False,
```
Comment (rt17603) [inline openghg/dataobjects/_searchresults.py:97]: Would be useful to have an explanation of additional inputs for the `retrieve()` function in the doc string - e.g. What does `elevate_inlet` mean?

**Review by rt17603 — APPROVED** [review]
I've added a few small comments of details to add, so would be nice if they could be updated, but other than that looks good. Lets get this in and then we can concentrate on the other PRs and issues.

File: openghg/dataobjects/_searchresults.py (hunk 2 above)
Comment (gareth-j) [reply openghg/dataobjects/_searchresults.py:97]: Done

File: openghg/retrieve/_search.py (hunk 1 above)
Comment (gareth-j) [reply openghg/retrieve/_search.py:164]: Done.

**Pull request merged**